deepchem/models/tests/test_pretrained.py

```diff
@@ -77,6 +78,63 @@ def test_load_from_pretrained(self):
       dest_val = dest_var.numpy()
       np.testing.assert_array_almost_equal(source_val, dest_val)
 
+  def test_load_pretrained_subclassed_model(self):
+    from rdkit import Chem
+    bi_tasks = ['a', 'b']
+    y = np.ones((3, 2))
+    smiles = ['C', 'CC', 'CCC']
+    mols = [Chem.MolFromSmiles(smile) for smile in smiles]
+    featurizer = dc.feat.ConvMolFeaturizer()
+    X = featurizer.featurize(mols)
+    dataset = dc.data.NumpyDataset(X, y, ids=smiles)
+
+    source_model = dc.models.GraphConvModel(
+        n_tasks=len(bi_tasks),
+        graph_conv_layers=[128, 128],
+        dense_layer_size=512,
+        dropout=0,
+        mode='regression',
+        learning_rate=0.001,
+        batch_size=8,
+        model_dir="model")
+    source_model.fit(dataset)
+
+    dest_model = dc.models.GraphConvModel(
+        n_tasks=len(bi_tasks),
+        graph_conv_layers=[128, 128],
+        dense_layer_size=512,
+        dropout=0,
+        mode='regression',
+        learning_rate=0.001,
+        batch_size=8)
+
+    X_b, y_b, w_b, ids_b = next(
+        dataset.iterbatches(batch_size=8, deterministic=True, pad_batches=True))
+    multiConvMol = ConvMol.agglomerate_mols(X_b)
+    n_samples = np.array(X_b.shape[0])
+    inputs = [
+        multiConvMol.get_atom_features(), multiConvMol.deg_slice,
+        np.array(multiConvMol.membership), n_samples
+    ]
+    for i in range(1, len(multiConvMol.get_deg_adjacency_lists())):
+      inputs.append(multiConvMol.get_deg_adjacency_lists()[i])
```

Comment:
This isn't for the current PR, but perhaps we should make `Model` classes responsible for generating their inputs. Typically classes do this in `default_generator`, but perhaps we can add a `Model.generate_input_batch(X_b, y_b, w_b, ids_b)` method. Then this API would smooth down quite nicely.